Control/Concurrent/Async/Internal.hs

```diff
@@ -95,40 +99,65 @@ compareAsyncs (Async t1 _) (Async t2 _) = compare t1 t2
 -- (see module-level documentation for details).
 --
 -- __Use 'withAsync' style functions wherever you can instead!__
-async :: IO a -> IO (Async a)
+async ::
+#ifdef DEBUG_AUTO_LABEL
+  HasCallStack =>
+#endif
```

Comment:
Would an alias work there? Like the following:

```haskell
#ifdef DEBUG_AUTO_LABEL
import qualified GHC.Stack
#else
import qualified GHC.Exts
#endif
...
#ifdef DEBUG_AUTO_LABEL
type HasCallStack = GHC.Stack.HasCallStack
#else
type HasCallStack = () :: GHC.Exts.Constraint
#endif
```
That way one won't need to put `ifdef` in every single type signature.

Reply:
Sure thing. I prioritized being explicit everywhere but what you suggest sounds good to me too.